src/Nexus/Extensibility/DataSource/DataSourceController.cs

```diff
@@ -350,7 +319,7 @@ internal class DataSourceController(
             .Where(readUnit => readUnit.CatalogItemRequest.BaseItem is null)
             .ToArray();
 
-        Logger.LogTrace("Load {RepresentationCount} original representations", originalReadUnits.Length);
+        _logger.LogTrace("Load {RepresentationCount} original representations", originalReadUnits.Length);
 
         var originalProgress = new Progress<double>();
         var originalProgressFactor = originalReadUnits.Length / (double)readUnits.Length;
```

Comment:
This PR should not touch the progress calculation algorithm but you found a bug: `originalProgressFactor` and `processingProgressFactor` both should actually be used but they are not. The algorithm does the following:

- The method `ReadAsync` gets a list of read requests (in the form of the ugly `CatalogItemRequestPipeWriter[]` parameter).
- Now the task is to calculate proper progress so the web UI shows the user something meaningful
- There are two type of read operations: 
  - read raw data (= `original`)
  - read to-be-processed data (= aggregated data: `mean`, `min`, `max`, etc), which will be calculated on the fly or read from a cache file if it exists
- The progress should be equal to the ratio of `original` vs `processing` read requests, i.e. when there is `1` original data read request and `2` processing requests, the progress should show `1/3` = `33%` when the original data read request has finished.
- The factors `originalProgressFactor` and `processingProgressFactor` exist to make this proportional progress happen and I forgot to apply them
- This will be solved with my next commit

Reply:
Yeah, I know, the change request is a bit outside of this PR's scope. This is a habit from my COAST days. When we started a new ticket we had an agreement that we clean up the code in the file we had to change/extend. The clean up rule applied to small changes e.g. removing unused code like using statements, variables or methods, or replacing string literals with constants etc.; for bigger changes a new ticket had to be created. I will try to suppress this old habit in future PRs. ;-) I am glad though that it lead to the discovery of a bug. Thanks for the explanation. :-)

Reply:
Please keep mentioning everything you discover, this is important and useful :-)